src/controller/config/controller.go

```diff
@@ -0,0 +1,217 @@
+	return ConvertForGet(ctx, configs, false)
+}
+
+func (c *controller) AllConfigs(ctx context.Context) (map[string]interface{}, error) {
+	configs := c.mgr.GetAll(ctx)
+	return configs, nil
+}
+
+func (c *controller) UpdateUserConfigs(ctx context.Context, conf map[string]interface{}) error {
+	err := c.mgr.Load(ctx)
+	if err != nil {
+		return err
+	}
+	isSysErr, err := c.validateCfg(ctx, conf)
+	if err != nil {
+		if isSysErr {
+			log.Errorf("failed to validate configurations: %v", err)
+			return fmt.Errorf("failed to validate configuration")
+		}
+		return err
+	}
+	if err := c.mgr.UpdateConfig(ctx, conf); err != nil {
+		log.Errorf("failed to upload configurations: %v", err)
+		return fmt.Errorf("failed to validate configuration")
+	}
+	return nil
+}
+
+func (c *controller) validateCfg(ctx context.Context, cfgs map[string]interface{}) (bool, error) {
+	flag, err := authModeCanBeModified(ctx)
+	if err != nil {
+		return true, err
+	}
+	if !flag {
+		if failedKeys := c.checkUnmodifiable(ctx, cfgs, common.AUTHMode); len(failedKeys) > 0 {
+			return false, errors.BadRequestError(nil).
+				WithMessage(fmt.Sprintf("the keys %v can not be modified as new users have been inserted into database", failedKeys))
+		}
+	}
+	err = c.mgr.ValidateCfg(ctx, cfgs)
+	if err != nil {
+		return false, errors.BadRequestError(err)
+	}
+	return false, nil
+}
+
+func (c *controller) checkUnmodifiable(ctx context.Context, cfgs map[string]interface{}, keys ...string) (failed []string) {
+	if c.mgr == nil || cfgs == nil || keys == nil {
+		return
+	}
+	for _, k := range keys {
+		v := c.mgr.Get(ctx, k).GetString()
+		if nv, ok := cfgs[k]; ok {
+			if v != fmt.Sprintf("%v", nv) {
+				failed = append(failed, k)
+			}
+		}
+	}
+	return
+}
+
+// ScanAllPolicy is represent the json request and object for scan all policy
+// Only for migrating from the legacy schedule.
+type ScanAllPolicy struct {
+	Type  string                 `json:"type"`
+	Param map[string]interface{} `json:"parameter,omitempty"`
+}
+
+// ConvertForGet - delete sensitive attrs and add editable field to every attr
+func ConvertForGet(ctx context.Context, cfg map[string]interface{}, internal bool) (map[string]*config.Value, error) {
+	result := map[string]*config.Value{}
+
+	mList := metadata.Instance().GetAll()
+
+	for _, item := range mList {
+		val, exist := cfg[item.Name]
+		// skip undefined items
+		if !exist {
+			continue
+		}
+
+		switch item.ItemType.(type) {
+		case *metadata.PasswordType:
+			// remove password for external api call
+			if !internal {
+				delete(cfg, item.Name)
+				continue
+			}
+		case *metadata.MapType, *metadata.StringToStringMapType:
+			// convert to string for map type
+			valByte, err := json.Marshal(val)
+			if err != nil {
+				return nil, err
+			}
+			val = string(valByte)
+		}
+		result[item.Name] = &config.Value{
+			Val:      val,
+			Editable: true,
+		}
+	}
+
+	// default value for ScanAllPolicy
+	if _, ok := result[common.ScanAllPolicy]; !ok {
```

Comment:
do we still need this? @heww

Reply:
I really don't know whether it's needed for us.

Reply:
keep it for now